### PR TITLE
Add HTMLHintBear

### PR DESCRIPTION
**Pull request by Asnelchristian**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "tslint": "~3",
     "typescript": ">=1.7.3",
     "ramllint": "~1.2.2",
-    "write-good": "~0.9.1"
+    "write-good": "~0.9.1",
+	"htmlhint": "~0.9"
```
Comment (gaocegege) [inline package.json]: There is a `\t`, not spaces. 😃

File: bears/hypertext/HTMLHintBear.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
+from coalib.bearlib.abstractions.Linter import linter
+from dependency_management.requirements.NpmRequirement import NpmRequirement
+
+
+@linter(
+    executable='htmlhint',
+    output_format='regex',
+    output_regex=r'(?P<filename>.+):(?P<line>\d+):(?P<column>\d+):\s*'
+                 r'(?P<message>.+)\s*(\d+)(\sproblems)\s*',
```
Comment (meetmangukiya) [inline bears/hypertext/HTMLHintBear.py]: Don't use capture groups other then ones coala uses.. the digit one and the `(\sproblems)`, btw is this regex correct? `\sproblems` will match `\nproblems`, `\tproblems`, etc. I don't think that is intended ?

File: bears/hypertext/HTMLHintBear.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    @staticmethod
+    def create_arguments(filename, file, config_file):
+        return (filename, '--config', config_file, '-f', 'unix')
+
+    @staticmethod
+    def generate_config(filename, file,
+                        enforce_lowercase_tagname: bool=True,
+                        enforce_lowercase_attribute: bool=True,
+                        allow_attribute_value_in_single_quote: bool=False,
+                        prohibit_empty_attribute: bool=False,
```
Comment (meetmangukiya) [inline bears/hypertext/HTMLHintBear.py]: Shouldn't this be true, IIUC it is for cases like `<p style=""></p>` ?

File: bears/hypertext/HTMLHintBear.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            Require that attributes value are set.
+        :param prohibit_attribute_duplication:
+            Require that the same attribute is not defined more that once in
+            a tag.
+        :param use_doctype_first:
+            Require that Doctype is first.
+        :param enforce_tag_pair:
+            Require that tags are paired.
+        :param enforce_self_close_empty_tag:
+            Require that empty must closed by self.
```
Comment (meetmangukiya) [inline bears/hypertext/HTMLHintBear.py]: Not quite clear :worried:

File: bears/hypertext/HTMLHintBear.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+           enforce_self_close_empty_tag=convert_bool_to_str(
+               enforce_self_close_empty_tag),
+           allow_duplicate_attribute_id=convert_bool_to_str(
+               not allow_duplicate_attribute_id),
+           require_title_tag=convert_bool_to_str(require_title_tag))
+        return configs
+
+
+def convert_bool_to_str(boolean):
+    result = 'true' if boolean else 'false'
```
Comment (meetmangukiya) [inline bears/hypertext/HTMLHintBear.py]: I suppose you meant to return this..

**Comment by meetmangukiya** [comment]
ping @Asnelchristian
ping @Asnelchristian

**Comment by yash-nisar** [comment]
Continued in https://github.com/coala/coala-bears/commit/b49553a8d62cae9989d0778728adb09fecb7fffb.

@Makman2 @Asnelchristian We need to close this. :wink:

**Pull request closed without merging**